test/test_datasets.py

```diff
@@ -1494,6 +1494,53 @@ def test_num_examples_test50k(self):
             assert len(dataset) == info["num_examples"] - 10000
 
 
+class MovingMNISTTestCase(datasets_utils.DatasetTestCase):
+    DATASET_CLASS = datasets.MovingMNIST
+    FEATURE_TYPES = (torch.Tensor,)
+
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(split=(None, "train", "test"), split_ratio=(10, 1, 19))
+
+    def inject_fake_data(self, tmpdir, config):
+        raw_dir = os.path.join(tmpdir, self.DATASET_CLASS.__name__)
+        os.makedirs(raw_dir, exist_ok=True)
+        num_samples = self._num_samples()
+        frame_size = self._frame_size()
+        data = np.concatenate(
+            [
+                np.zeros((config["split_ratio"], num_samples, *frame_size)),
+                np.ones((20 - config["split_ratio"], num_samples, *frame_size)),
+            ]
+        )
+        np.save(os.path.join(raw_dir, "mnist_test_seq.npy"), data)
+        return num_samples
+
+    @datasets_utils.test_all_configs
+    def test_split(self, config):
+        if config["split"] is None:
+            return
+
+        num_samples = self._num_samples()
+        frame_size = self._frame_size()
+        with self.create_dataset(config) as (dataset, info):
+            print(dataset.data.shape)
+            print(torch.zeros(config["split_ratio"], num_samples, *frame_size).shape)
+            if config["split"] == "train":
+                assert torch.equal(dataset.data, torch.zeros(num_samples, config["split_ratio"], 1, *frame_size))
```

Comment:
Instead of checking against a zero vector, can't we simply check if all elements are zero?

```suggestion
            if config["split"] == "train":
                assert (dataset.data == 0).all()
```

Same for the test split as well.